public/app/plugins/panel/gauge/GaugePanel.tsx

```diff
@@ -81,6 +82,21 @@ export class GaugePanel extends PureComponent<PanelProps<Options>> {
   render() {
     const { height, width, data, renderCounter, options } = this.props;
 
+    // We could make this not hard coded but instead dependent on panel dimensions / number of gauges
+    const GAUGE_MIN_HEIGHT = 200;
+    const GAUGE_MIN_WIDTH = 200;
+
+    // calculate if there is overflow
+    const numberOfGauges = this.getValues().length;
+
+    const remainderWidth = width - numberOfGauges * GAUGE_MIN_WIDTH;
+    const isHorizontalOverflow = options.orientation === VizOrientation.Horizontal && remainderWidth < 0;
+
+    const remainderHeight = height - numberOfGauges * GAUGE_MIN_HEIGHT;
+    const isVerticalOverflow = options.orientation === VizOrientation.Vertical && remainderHeight < 0;
+
+    const isOverflow = isHorizontalOverflow || isVerticalOverflow;
+
```

Comment:
can't this logic be inside the VizRepeater instead? it knows the number of elements, and you pass GAUGE_MIN_WIDTH  via minVizWidth (same with height), it knows the orientation etc. 

The big problem (beyond having to duplicate it for bar gauge) with having it here is needing that additional call to getValues